src/xrpld/app/misc/WamrVM.cpp

```diff
@@ -640,6 +640,62 @@ WamrEngine::call(
     return call<NR>(func, in, p.data(), p.size(), std::forward<Types>(args)...);
 }
 
+Expected<int32_t, NotTEC>
+WamrEngine::preflight(
+    wbytes const& wasmCode,
+    std::string_view funcName,
+    std::vector<WasmImportFunc> const& imports,
+    std::vector<WasmParam> const& params)
+{
+    try
+    {
+        return preflightHlp(wasmCode, funcName, imports, params);
+    }
+    catch (std::exception const& e)
+    {
+        std::cerr << "WAMR: preflight exception: " << e.what() << std::endl;
+    }
+    catch (...)
+    {
+        return temUNKNOWN;
+    }
+    return Unexpected<NotTEC>(temBAD_WASM);
+}
+
+Expected<int32_t, NotTEC>
+WamrEngine::preflightHlp(
+    wbytes const& wasmCode,
+    std::string_view funcName,
+    std::vector<WasmImportFunc> const& imports,
+    std::vector<WasmParam> const& params)
+{
+    // Create and instantiate the module.
+    if (!wasmCode.empty())
+    {
+        int const m = addModule(wasmCode, true, imports);
+        if (m < 0)
+            return temUNKNOWN;
+    }
+
+    if (!module)
+        return temUNKNOWN;
+
+    // Call main
+    auto* f = getFunc(!funcName.empty() ? funcName : "_start");
```

Comment:
It should return error, cause this is check.

Reply:
That's caught in the `preflight` function